jsonpb/decode.go

```diff
@@ -390,6 +391,20 @@ func isSingularWellKnownValue(fd protoreflect.FieldDescriptor) bool {
 	return false
 }
 
+func isUnmarshalJSONPB(v protoreflect.Value, fd protoreflect.FieldDescriptor) bool {
```

Comment:
Please call this `isSingularJSONPBUnmarshaler` for consistency with `isSingularWellKnownValue` and please declare this function after `isSingularWellKnownValue`.

Also, please make the body of this similar to `isSingularWellKnownValue`:
```go
if fd.Message() != nil && fd.Cardinality() != protorflect.Repeated {
	_, ok := proto.MessageV1(v.Interface()).(JSONPBUnmarshaler)
	return ok
}
return false
```